backend/aml/development_settings.py

```diff
@@ -25,6 +26,26 @@
 
     MIDDLEWARE += ["debug_toolbar.middleware.DebugToolbarMiddleware"]
 
+
+def is_debugpy_running(port):
+    """Check if debugpy is already running on the specified port."""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(("127.0.0.1", port)) == 0
+
+
+if DEBUG:
+    try:
+        import debugpy
+
+        if not is_debugpy_running(5678):
+            print("Starting debugpy on port 5678 🐛")
+            debugpy.listen(("0.0.0.0", 5678))
+        else:
+            print("Debugpy is already running on port 5678 👍")
```

Comment:
This check has been added since the django server (but not the container) will restart once you change a file. The debug server will spawn a separate process as far as I understood and won't get terminated whenever the django server restarts.